element_array_ephys/export/nwb/nwb.py

```diff
@@ -0,0 +1,600 @@
+
+    for lfp_record in (ephys.LFP & session_key).fetch(as_dict=True):
+        probe_id = (ephys.ProbeInsertion & lfp_record).fetch1("probe")
+
+        lfp_electrodes_query = ephys.LFP.Electrode & lfp_record
+        lfp_data = LFPDataChunkIterator(lfp_electrodes_query)
+
+        nwb_lfp.create_electrical_series(
+            name=f"ElectricalSeries{lfp_record['insertion_number']}",
+            description=f"LFP from probe {probe_id}",
+            data=H5DataIO(lfp_data, compression=True),
+            timestamps=lfp_record["lfp_time_stamps"],
+            electrodes=nwbfile.create_electrode_table_region(
+                name="electrodes",
+                description="electrodes used for LFP",
+                region=[
+                    mapping[(probe_id, x)]
+                    for x in lfp_electrodes_query.fetch("electrode")
+                ],
+            ),
+        )
+
+
+def add_ephys_lfp_from_source_to_nwb(
+    session_key: dict, ephys_root_data_dir, nwbfile: pynwb.NWBFile, end_frame=None):
+    """
+    Read the LFP data directly from the source file. Currently, only works for SpikeGLX data.
+
+    ephys.EphysRecording::recording_datetime -> acquisition
+
+    Parameters
+    ----------
+    session_key: dict
+    nwbfile: pynwb.NWBFile
+    end_frame: int, optional
+        use for small test conversions
+
+    """
+    if nwbfile.electrodes is None:
+        add_electrodes_to_nwb(session_key, nwbfile)
+
+    mapping = get_electrodes_mapping(nwbfile.electrodes)
+
+    ecephys_module = get_module(
+        nwbfile, name="ecephys", description="preprocessed ephys data"
+    )
+
+    lfp = pynwb.ecephys.LFP()
+    ecephys_module.add(lfp)
+
+    for ephys_recording_record in (ephys.EphysRecording & session_key).fetch(
+        as_dict=True
+    ):
+        probe_id = (ephys.ProbeInsertion() & ephys_recording_record).fetch1("probe")
+
+        relative_path = (
+            ephys.EphysRecording.EphysFile & ephys_recording_record
+        ).fetch1("file_path")
+        relative_path = relative_path.replace("\\","/")
+        file_path = find_full_path(ephys_root_data_dir, relative_path)
+
+        if ephys_recording_record["acq_software"] == "SpikeGLX":
+            extractor = extractors.read_spikeglx(os.path.split(file_path)[0], "imec.lf")
+        else:
+            raise ValueError(
+                f"unsupported acq_software type: {ephys_recording_record['acq_software']}"
+            )
+
+        if end_frame is not None:
+            extractor = extractor.frame_slice(0, end_frame)
+
+        recording_channels_by_id = (
+            probe.ElectrodeConfig.Electrode() & ephys_recording_record
+        ).fetch("electrode")
+
+        conversion_kwargs = gains_helper(extractor.get_channel_gains())
+
+        lfp.add_electrical_series(
+            pynwb.ecephys.ElectricalSeries(
+                name=f"ElectricalSeries{ephys_recording_record['insertion_number']}",
+                description=str(ephys_recording_record),
```

Comment:
```suggestion
                description=f"LFP from probe {probe_id}",
```

Reply:
Am mirroring the `add_ephys_lfp_from_dj_to_nwb` function.  But we can keep it as is.